src/backend/quarkus-management-api/src/main/java/com/nttdata/quarkus/management/api/controller/ContactsResource.java

```diff
@@ -0,0 +1,101 @@
+package com.nttdata.quarkus.management.api.controller;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import javax.ws.rs.*;
+
+import com.nttdata.quarkus.management.api.model.api.Contact;
+import com.nttdata.quarkus.management.api.model.database.Contacts;
+import com.nttdata.quarkus.management.api.service.ContactsService;
+import org.jboss.resteasy.annotations.jaxrs.PathParam;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Path("/management")
+@ApplicationScoped
+public class ContactsResource {
+
+    @Inject
+    ContactsService contactsService;
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/contacts")
+    @Transactional
+    public List<Contact> getContacts() {
+
+        return contactsService.getContacts().stream()
+                .map(this::mapContact)
+                .collect(Collectors.toList());
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/contacts/{contactId}")
+    @Transactional
+    public Contact getContact(@PathParam BigInteger contactId) {
+        return mapContact(contactsService.getContact(contactId));
+    }
+
+    @POST
+    @Path("/contacts/add")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Transactional
+    public Response create(Contact contact) {
+
+        if (contact.getContactId() != null) {
+            throw new WebApplicationException("Id not required.", 422);
```

Comment:
shouldnt the message here say `contact you are trying to add already exists` or something along those lines?

Reply:
It isn't actually checking the existence this is more to prevent unintended call to add